maint: bump up version of @patternfly/* npm modules

KKoukiou opened the pull request:
This was a breaking release which changed:

* directory structure in the @patternfly/patternfly repository thus
includes to patternfly-variables has to be adjusted
* Renamed/Removed components from @patternfly/react-core
* @patternfly/react-core now depends on react-popper:2.2.3 which is
already brought as dependency of patternfly-react npm module (PF3)
thus we pin it here in the packag.json to avoid it being pulled in the
old version

KKoukiou commented:
@garrett there is a remaining issue I need some help with:
![Screen Shot 2020-07-14 at 19 21 09](https://user-images.githubusercontent.com/14921356/87456725-a267be00-c607-11ea-9fe8-9699da5d551d.png)


Aside from that multiple things change design including the following:
Inline notifications: 
![Screen Shot 2020-07-14 at 19 25 15](https://user-images.githubusercontent.com/14921356/87456826-c3301380-c607-11ea-86c7-4b3e773590e5.png)


Navigation:
![Screen Shot 2020-07-14 at 19 25 53](https://user-images.githubusercontent.com/14921356/87456915-e064e200-c607-11ea-8940-757db0b3da97.png)

KKoukiou commented:
The typeaheadselect component is crashing with this upgrade. Seems like popper.js npm dependency is conflicty between the versions, since new PF4 needs different version than patternfly-react.
Typeahead should be replaced anyway with PF4 so I will try to implement that within this PR instead of debugging the broken dependency tree.

garrett commented:
We're getting closer.

I pushed up some more fixes for the list.

We should probably look at switching the sudo banner to an actual banner. I also need to adjust the navigation a little more. (There the hover state is missing the thin line, for example. I need to look a bit closer otherwise too.)

Once we have the tabs and the select swapped out and the style fixes mentioned above, I think we'll be ready. (I guess we might be able to land this next week at some point?)

KKoukiou commented:
@garrett I fixed the TypeAheadSelect
For the Nav I did not change it to Tabs component, but I adjusted the CSS and now it looks almost like previously.

I tried the Tabs component, but it's really not simple CSS overrides to make it work. We should change it but this PR has grown too much anyway. :tipping_hand_woman: 

Short version of the story is that the `Tabs` components have their content as children in the dom tree, whilst `Nav` component does not.
In our customized CSS in cockpit we apply `display: flex` to the `Nav` header so that the tabs/nav appear in the same line with possibly existing action buttons.
Of course that messes up with the content of the Tabs.

So, let's fix the Nav -> Tabs in another PR because it's not trivial.

martinpitt commented on pkg/lib/cockpit-components-file-autocomplete.jsx:
```diff
@@ -113,78 +112,67 @@ export class FileAutoComplete extends React.Component {
+                          className={option.type}
+                          value={{
+                              ...option,
+                              toString: function() { return this.path },
```
This looks odd -- doesn't function() introduce a new `this`? I.e. shouldn't this be `toString: () => this.path`?

There are also some more toString() methods in this PR with the same issue.

KKoukiou replied:
That's exactly what I want - the new `this` from the context of the class that this function is going to be used at at the side of the @patternfly/react-core code.

KKoukiou replied:
So this is here on the purpose - the arrow notation would break the functionality.

martinpitt replied:
Ack, thanks!

martinpitt commented on pkg/lib/cockpit-components-file-autocomplete.jsx:
```diff
@@ -113,78 +112,67 @@ export class FileAutoComplete extends React.Component {
+                variant={SelectVariant.typeahead}
+                id={this.props.id}
+                placeholderText={placeholder}
+                noResultsFoundText={cockpit.format("No such file or directory '$0'", this.state.value)}
```
This needs i18n. Not a new bug, but let's fix it while we are at it.

martinpitt commented on pkg/machines/components/create-vm-dialog/createVmDialog.jsx:
```diff
@@ -19,8 +19,11 @@
-import { Button, Tooltip, TooltipPosition } from '@patternfly/react-core';
+import { FormGroup, HelpBlock, Modal } from 'patternfly-react';
+import {
+    Select as SelectTypeAhead, SelectOption, SelectVariant,
```
Why the SelectTypeAhead aliasing? IMHO that's rather confusing when reading the code and trying to map it to the PF4 documentation. I see it's needed to disambiguate from cockpit-components-select, but maybe call them PFSelect and CockpitSelect?

KKoukiou replied:
I needed the Aliasing because we already include the Select component from another file. I will replace the strings to the suggested naming.

martinpitt commented:
This [TestPages.testBasic](https://logs.cockpit-project.org/logs/pull-14358-20200721-100255-f8c3109f-rhel-8-3-distropkg/log.html#256-1) failure on rhel-8-3-distropkg is a bit stubborn:

    RuntimeError: ReferenceError: jQuery is not defined

and then in the retry

    systemctl start test.timer

Retrying once more to compare logs, I didn't keep the previous one.
